src/main/scala/org/constellation/invertedmap/InvertedMap.scala

```diff
@@ -0,0 +1,54 @@
+package org.constellation.invertedmap
+
+import scala.collection.generic.{CanBuildFrom, GenMapFactory}
+import scala.collection.immutable.MapLike
+import scala.collection.{GenTraversableOnce, mutable}
+
+final class InvertedMap[A, B] private (m: Map[B, Set[A]])
+    extends Map[A, B]
+    with MapLike[A, B, InvertedMap[A, B]]
+    with Serializable {
+
+  override protected[this] def newBuilder: mutable.Builder[(A, B), InvertedMap[A, B]] =
+    InvertedMap.newBuilder[A, B]
+
+  override def size: Int = Inverted.size(m)
+
+  def this() = this(Map.empty[B, Set[A]])
```

Comment:
Why is this() returning an empty InvertedMap?

Reply:
It's used there:
```
def empty[A, B] = new InvertedMap[A, B]()
```

Reply:
Maybe just embed it in `def empty[A, B]...`?
I'm not sure that is an expected behaviour for `this`.

Reply:
yes, it's used in the same way in `TreeMap` or `Stack`